File: common/src/main/java/io/github/xiewuzhiying/vs_addition/compats/computercraft/peripherals/CheatFlapBearingPeripheral.java
package io.github.xiewuzhiying.vs_addition.compats.computercraft.peripherals;

import dan200.computercraft.api.lua.LuaFunction;
import net.minecraft.core.BlockPos;
import net.minecraft.world.level.Level;
import org.valkyrienskies.clockwork.content.contraptions.flap.FlapBearingBlockEntity;

public class CheatFlapBearingPeripheral extends FlapBearingPeripheral{
    public CheatFlapBearingPeripheral(String type, FlapBearingBlockEntity tileEntity, Level level, BlockPos blockPos) {
        super(type, tileEntity, level, blockPos);
    }

    @LuaFunction(mainThread = true)
    public final Object setAngle(double angle){
        if(this.tileEntity.isRunning()) {
            this.tileEntity.setAngle((float) angle);
            return true;
        }
        return false;
    }
}

File: common/src/main/kotlin/io/github/xiewuzhiying/vs_addition/compats/computercraft/peripheral/CheatFlapBearingPeripheral.kt
package io.github.xiewuzhiying.vs_addition.compats.computercraft.peripheral

import dan200.computercraft.api.lua.LuaFunction
import io.github.xiewuzhiying.vs_addition.mixinducks.vs_clockwork.flap_bearing.FlapBearingBlockEntityMixinDuck
import org.valkyrienskies.clockwork.content.contraptions.flap.FlapBearingBlockEntity

class CheatFlapBearingPeripheral(peripheralType: String, tileEntity: FlapBearingBlockEntity) : FlapBearingPeripheral(peripheralType,
    tileEntity
) {
    @LuaFunction(mainThread = true)
    override fun setAngle(angle: Double): Boolean {
        if (tileEntity.isRunning) {
            (this.tileEntity as FlapBearingBlockEntityMixinDuck).lockedFlapAngle = angle.toFloat()
            tileEntity.setAngle(angle.toFloat())
            (this.tileEntity as FlapBearingBlockEntityMixinDuck).isLocked = true
            return true
        }
        return false
    }
}